src/deadlock-detection/threaddeadlock.h

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2019 Greg Griffith
+// Copyright (c) 2019 The Bitcoin Unlimited developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_THREAD_DEADLOCK_H
+#define BITCOIN_THREAD_DEADLOCK_H
+
+#include <boost/lexical_cast.hpp>
+#include <boost/thread.hpp>
+#include <memory>
+#include <mutex>
+
+#include "lockorder.h"
+#include "utilstrencodings.h"
+
+#ifdef DEBUG_LOCKORDER // this ifdef covers the rest of the file
+#ifdef __linux__
+#include <sys/syscall.h>
+inline uint64_t getTid(void)
+{
+    // "native" thread id used so the number correlates with what is shown in gdb
+    pid_t tid = (pid_t)syscall(SYS_gettid);
+    return tid;
+}
+#else
+#include <functional>
+inline uint64_t getTid(void)
+{
+    // Note: there is no guaranteed way to turn the thread-id into an int
+    // since it's an opaque type. Just about the only operation it supports
+    // is std::hash (so that thread id's may be placed in maps).
+    // So we just do this.
+    static std::hash<std::thread::id> hasher;
+    return uint64_t(hasher(std::this_thread::get_id()));
+}
+#endif
+
+struct LockData
+{
+    // Very ugly hack: as the global constructs and destructors run single
```

Comment:
Where is this happening?  Since you've placed lockdata first in globals.cpp it should be destructed last.  perhaps some other criticalsections need to be moved to globals.cpp?

Reply:
i believe this was a relic from before i moved it into globals.cpp. i have removed this data member.